### PR TITLE
Add C sample

### PR DESCRIPTION
**Pull request by DavidAylaian**
Sample was incorrectly being identified as C++

### CODE REVIEW

File: samples/C/asm.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* CarbonOS System/Kernel
+ * Copyright 2015-2017 David Aylaian
+ * Licensed under GitHub CLA: https://cla.github.com/agreement
```
Comment (pchaigno) [inline samples/C/asm.h]: I don't think you can license a file under a CLA...

Comment (DavidAylaian) [inline samples/C/asm.h]: I'll put it under the unlicense then.

Comment (Alhadis) [inline samples/C/asm.h]: Well, that works, but a [permissive license](https://github.com/github/linguist/blob/9289704/vendor/licenses/config.yml#L2-L11) (e.g., MIT/Expat-compatible) is really all we need. =) It is of course your personal choice, however.

Comment (Alhadis) [inline samples/C/asm.h]: I also noticed that the project the code was originally authored for is [licensed under the Apache 2.0 License](https://github.com/DavidAylaian/CarbonOS/blob/6dbd6ec/LICENSE.md). Which, [as you'll see](https://github.com/github/linguist/blob/master/vendor/licenses/config.yml#L2), is an acceptably permissive license. =)

In other words, you didn't have to relicense anything. ;-)

Comment (DavidAylaian) [inline samples/C/asm.h]: @Alhadis Ok, sounds great. I wasn't sure if Apache 2.0 was an acceptably permissive license.

**Pull request merged**